contracts/src/NativeTokenDestination.sol

```diff
@@ -0,0 +1,321 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: Ecosystem
+
+pragma solidity 0.8.18;
+
+import {
+    TeleporterTokenDestination,
+    TeleporterFeeInfo,
+    TeleporterMessageInput
+} from "./TeleporterTokenDestination.sol";
+import {Address} from "@openzeppelin/contracts@4.8.1/utils/Address.sol";
+import {INativeMinter} from
+    "@avalabs/subnet-evm-contracts@1.2.0/contracts/interfaces/INativeMinter.sol";
+import {INativeTokenDestination} from "./interfaces/INativeTokenDestination.sol";
+import {TeleporterOwnerUpgradeable} from "@teleporter/upgrades/TeleporterOwnerUpgradeable.sol";
+// We need IAllowList as an indirect dependency in order to compile.
+// solhint-disable-next-line no-unused-import
+import {IAllowList} from "@avalabs/subnet-evm-contracts@1.2.0/contracts/interfaces/IAllowList.sol";
+import {IWrappedNativeToken} from "./interfaces/IWrappedNativeToken.sol";
+import {SendTokensInput} from "./interfaces/ITeleporterTokenBridge.sol";
+
+/**
+ * THIS IS AN EXAMPLE CONTRACT THAT USES UN-AUDITED CODE.
+ * DO NOT USE THIS CODE IN PRODUCTION.
+ */
+
+/**
+ * @notice Implementation of the {INativeTokenDestination} interface.
+ *
+ * @dev This contract pairs with exactly one `TokenSource` contract on the source chain.
+ * It mints and burns native tokens on the destination chain corresponding to locks and unlocks on the source chain.
+ */
+contract NativeTokenDestination is
```

Comment:
WIll there be a `WrappedNativeTOken` contract we need to deploy with the `NativeTokenDestination`?